### ISSUE TITLE
Edit JSON Schema dialog triggers save event by every key press, which triggers git commit

### ISSUE THREAD

**Issue by chrda81** [sartography/spiff-arena#1636]
Hi,

I configured the setting `SPIFFWORKFLOW_BACKEND_GIT_COMMIT_ON_SAVE = true` yesterday and now see many commit messages for the same file in a short time period. My investigation shows that a save event of the "Edit JSON Schema" dialog is triggered by every key press, which then trigges a git commit. This makes the git history very hard to read.

I try to find an open issue for that case, so hopefully this is not a duplicate?

![image](https://github.com/sartography/spiff-arena/assets/36750848/2f7c5c7c-71f4-4297-a1da-0ca78f369ecb)

**Comment by burnettk** [comment]
Yeah, it is designed to autosave. I can’t remember how the debounce is configured. We could increase it  a bit so if doesn’t save as frequently.

**Comment by chrda81** [comment]
Hmm. If I could make a wish I would suggest to let the user decide, if autosave or manual save shall be used for this dialog :D
This could be realized with relative "less" effort in my opinon:

![image](https://github.com/sartography/spiff-arena/assets/36750848/f4dd2c5e-21c0-4e89-bed0-6f0bd18f4c08)

- add an extra param (e.g. boolean "manualSave") to the ReactFormBuilder
- add this param to the useEffect conditions to suppress the saving
- add the call to the backend to the closing method of the dialog

![image](https://github.com/sartography/spiff-arena/assets/36750848/ca3581f2-cf27-4daa-ae7c-31dbafcedf1c)

I'm still trying to get building process of the frontend code working, so I cannot make the changes and test them :( With the python backend it was relative easy to test locally.

**Comment by burnettk** [comment]
Here’s your frontend setup steps, fwiw: https://github.com/sartography/spiff-arena?tab=readme-ov-file#frontend-setup-local

**Comment by burnettk** [comment]
hoping you are not still getting hosed by this. feel free to re-open if needed.
